mlrun/runtimes/pod.py

```diff
@@ -260,7 +258,14 @@ def to_dict(self, fields=None, exclude=None, strip=False):
         return struct
 
     def apply(self, modify):
-        return apply_kfp(modify, self._cop, self)
+
+        # suppress kubeflow pipelines op registration in apply()
+        old_op_handler = _container_op._register_op_handler
+        _container_op._register_op_handler = lambda x: self.metadata.name
+        cop = ContainerOp("name", "image")
+        _container_op._register_op_handler = old_op_handler
```

Comment:
Your comment explains what you're doing, while the interesting information is why you're doing that
Also, this can very easily break if we upgrade kfp version, I suggest having a specific test for that (with a clear explanation why are we testing this and what is the risk) that would fail if for example kfp will change the signature of `_register_op_handler`

Reply:
added comments, we have `apply()` in many tests which will fail if kfp changes, so do we need another test?